### PR TITLE
Add Concox CRX1-New Protocol

### PR DESCRIPTION
**Pull request by JucianoC**
Add Concox CRX1-New Protocol
Reference: https://drive.google.com/file/d/0B7dIYsEOmrGASEtmT2FPWnk5U3c/view?usp=sharing

### CODE REVIEW

File: scapy/contrib/concox.py [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+from scapy.fields import XByteField
+from scapy.fields import XByteEnumField
+from scapy.fields import ByteEnumField
+from scapy.fields import StrFixedLenField
+from scapy.fields import ConditionalField
+from scapy.fields import FlagsField
+from scapy.fields import ByteField
+from scapy.fields import IntField
+from scapy.fields import XIntField
+from scapy.fields import StrLenField
```
Comment (gpotter2) [inline scapy/contrib/concox.py]: Please unify imports

File: scapy/contrib/concox.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+    0x01: "Chinese",
+    0x02: "English",
+}
+
+
+class BCDStrFixedLenField(StrFixedLenField):
+    def i2h(self, pkt, x):
+        if isinstance(x, bytes):
+            return binascii.b2a_hex(x)
+        return binascii.a2b_hex(x)
```
Comment (gpotter2) [inline scapy/contrib/concox.py]: Couldn't you use `XStrFixedLenField` ?

File: scapy/contrib/concox.py [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+                           None,
+                           CRX1NewPacketContent,
+                           length_from=lambda pkt: pkt.extended_packet_length),
+            lambda pkt: pkt.start_bit == 0x7979),
+        XShortField('end_bit', 0x0d0a),
+    ]
+
+
+bind_layers(TCP, CRX1New)
+bind_layers(UDP, CRX1New)
```
Comment (gpotter2) [inline scapy/contrib/concox.py]: This will break Scapy. You should have a port, or something to chose between this layer and others

**Comment by gpotter2** [comment]
Please also add unit tests :-)

Thanks for the PR !

File: scapy/contrib/concox.py (hunk 3 above)
Comment (guedou) [reply scapy/contrib/concox.py]: According to Google the port is 8821.

**Comment by guedou** [comment]
PR #2373 includes the requested fixes.

File: scapy/contrib/concox.py (hunk 3 above)
Comment (JucianoC) [reply scapy/contrib/concox.py]: Actually, you can configure any port in the device

Comment (JucianoC) [reply scapy/contrib/concox.py]: The manufacturer does not specify a default port in the protocol or in the manual.

Comment (gpotter2) [reply scapy/contrib/concox.py]: Binding literally all UDP and TCP packets to CRX doesn't seem like a viable solution either

Comment (JucianoC) [reply scapy/contrib/concox.py]: Of course, it's just to keep the register that the set port is unofficial.

**Pull request closed without merging**